### PR TITLE
Feature/issue 1

### LINKED ISSUE #1: Add extra argument to cmdstan to support clamped params
Add an extra argument to cmdstan to allow specifying a clamped parameter file

Something like this should work:

```
./mymodel sampling data file=blah.json clamped_file=clamp.json
```

Clamped file is optional, so handle it like inits that are strings

### PR DESCRIPTION
**Pull request by bbbales2**
#### Submisison Checklist

- [x] Run tests: `./runCmdStanTests.py src/test`
- [x] Declare copyright holder and open-source license: see below

#### Summary:

Fixes #1 

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company): Columbia University

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: src/cmdstan/arguments/arg_clamped_params.hpp [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+#include <cmdstan/arguments/singleton_argument.hpp>
+#include <string>
+
+namespace cmdstan {
+
+class arg_clamped_params : public string_argument {
+ public:
+  arg_clamped_params() : string_argument() {
+    _name = "clamped_params";
+    _description = std::string("File with clamped parameters");
```
Comment (Dashadower) [inline src/cmdstan/arguments/arg_clamped_params.hpp]: More "descriptive" description is strongly recommended :)

File: src/cmdstan/command.hpp [hunk 2]
```diff
@@ -255,6 +257,11 @@ int command(int argc, const char *argv[]) {
   }
   std::shared_ptr<stan::io::var_context> init_context = get_var_context(init);
 
+  // Read clamped parameters file
+  std::string clamped_params_filename
+      = dynamic_cast<string_argument *>(parser.arg("clamped_params"))->value();
+  std::shared_ptr<stan::io::var_context> clamped_params_context = get_var_context(clamped_params_filename);
```
Comment (Dashadower) [inline src/cmdstan/command.hpp:263]: Do all other references to this pointer in stan use a smart pointer?

**Review by Dashadower — DISMISSED** [review]

File: src/cmdstan/command.hpp (hunk 2 above)
Comment (bbbales2) [reply src/cmdstan/command.hpp:263]: I think this is the only pointer. Everything else is a reference to the value here.

Comment (Dashadower) [reply src/cmdstan/command.hpp:263]: Then shouldn't a unique_ptr be better?

Comment (bbbales2) [reply src/cmdstan/command.hpp:263]: Oh, you're right it's a simpler way to solve the same problem. I'll change these to unique ptrs later if I mess with this pull any.

Comment (bbbales2) [reply src/cmdstan/command.hpp:263]: I tried unique_ptrs and they worked but then got scared and switched back cause we're returning them from functions and was worried the destructor might get called or something.

**Comment by bbbales2** [comment]
Alright I went and certify it is Truly Annoying to Test this so just merging as is

**Pull request merged**